src/components/Home.js

```diff
@@ -0,0 +1,161 @@
+import React from "react";
+import {connect} from 'react-redux';
+import {withRouter} from 'react-router';
+import {getOffers,getCategories} from '../actions/homeActions';
+import {resetCartData} from '../actions/productActions';
+import SvgLoading from '../presentations/SvgLoading';
+import Header from '../presentations/Header';
+import Footer from '../presentations/Footer';
+import {getViewPortDimensions} from '../utils';
+import LazyLoad, { forceCheck } from 'react-lazyload';
+import Slider from 'react-slick';
+import { Modal } from "react-responsive-modal";
+import Cart from '../presentations/Cart';
+import Constants from '../constants';
+import toastr from 'toastr';
+class Home extends React.Component {
+    constructor(props){
+        super(props);
+        this.state={
+          currentInstance: this,
+          bannerData: [],
+          categoryData: [],
+          dimensions:{},
+          openModal: false,
+          forceCheckSliderImage: false
+        }
+        // this.navigateToPlp = this.navigateToPlp.bind(this);
+        // this.showCartView = this.showCartView.bind(this);
+        // this.onCloseModal = this.onCloseModal.bind(this);
+        // this.checkoutComplete = this.checkoutComplete.bind(this);
+        // this.forceCheckSliderImage = this.forceCheckSliderImage.bind(this);
+    }
+    componentDidMount(){
+      this.props.getOffers();
+      this.props.getCategories();
+    }
+    static getDerivedStateFromProps(props,state){
+      if(props.homeApi.bannerData && props.homeApi.bannerdata_searching_success){
+        state.bannerData = props.homeApi.bannerData;
+        state.currentInstance.forceCheckSliderImage();
+      }
+      if(props.homeApi.categoryData && props.homeApi.categorydata_searching_success){
+        state.categoryData = props.homeApi.categoryData;
+      }
+      return state;
+    }
+    forceCheckSliderImage=()=>{
+      forceCheck();
+    }
+    doResizeActions=()=>{
+      let dimensions = getViewPortDimensions();
+      this.setState({dimensions: dimensions});
+    }
+  checkoutComplete = () => {
+    this.props.resetCartData();
+    this.onCloseModal();
+    toastr.success(Constants.TEXTS.TOASTS.cart_emptied,Constants.TEXTS.TOASTS.order_placed,{timeOut:1000});
+  }
+  showCartView = () => {
+    this.setState({ openModal: true });
+    document.getElementsByTagName('body')[0].style.overflow = 'hidden';
+    document.getElementsByTagName('html')[0].style.overflow = 'hidden';
+  }
+
+  onCloseModal = () => {
+    this.setState({ openModal: false });
+    document.getElementsByTagName('body')[0].style.overflow = 'auto';
+    document.getElementsByTagName('html')[0].style.overflow = 'auto';
+  };
+    navigateToPlp=(e,data)=>{
+      if(data)
+      this.props.router.push('/products?cat='+data.name+'&id='+data.id);
```

Comment:
that's not how string literals are created

Reply:
changed mapping of string literals as I know here.